lib/provider/language_provider.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class LanguageProvider with ChangeNotifier {
+  Locale _currentLocale = const Locale('ar', 'SA');
+
+  LanguageProvider() {
+    loadLocale();
+  }
+
+  Locale get currentLocale => _currentLocale;
+
+  void changeLanguage(Locale locale) async {
+    if (_currentLocale == locale) return;
+    if (locale == const Locale("ar", "SA")) {
+      _currentLocale = const Locale("ar", "SA");
+      await _saveLocale("ar", "SA");
+    } else {
+      _currentLocale = const Locale("en", "US");
+      await _saveLocale("en", "US");
+    }
+    notifyListeners();
+  }
```

Comment:
Code can be optimized

```suggestion
void changeLanguage(Locale locale) async {
  if (_currentLocale != locale) {
    _currentLocale = locale;
    await _saveLocale(locale.languageCode, locale.countryCode);
    notifyListeners();
  }
}
```

Please verify.

Reply:
done ..